### PR TITLE
update upper bound for MarkupSafe

### LINKED ISSUE #19761: removing upper bound for markupsafe
### Description

Per discussion and guidance from #19753, opening this issue and PR https://github.com/apache/airflow/pull/19760 for review. Based on if all the tests pass, this could be reviewed further.

### Use case/motivation

Currently Jinja2 upper bound is jinja2>=2.10.1,<4 at https://github.com/apache/airflow/blob/main/setup.cfg#L121 as part of #16595

Jinja2 seems to require MarkupSafe>=2.0 at https://github.com/pallets/jinja/blob/main/setup.py#L6 as part of pallets/jinja@e2f673e

Based on this, is it feasible to consider updating upper bound in airflow for markupsafe>=1.1.1, <2.0 to allow 2.0 at https://github.com/apache/airflow/blob/main/setup.cfg#L126 ?

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by tsingh2k15**
Per discussion and guidance from #19753, opening this PR for review. Based on if all the tests pass, this could be reviewed further. Resolves #19761.

### CODE REVIEW

**Comment by potiuk** [comment]
Looks like 

```
tests/www/views/test_views_rendered.py::test_user_defined_filter_and_macros_raise_error: AssertionError
```

Is the one that should be looked at - other errors are intermittent, but this one looks like related.

**Comment by potiuk** [comment]
You can easily reproduce it in Breeze - (see the error log for instructions - you can also build your own image with `breeze --upgrade-to-newer-dependencies` in your PR and you should be able to reproduce it (and fix hopefully).

**Comment by uranusjr** [comment]
Weird, I just checked out this locally and can’t reproduce. Let me merge main and see if it’s just something wrong in the CI workers.

**Comment by uranusjr** [comment]
Oh never mind I got it.

**Comment by uranusjr** [comment]
I pushed a change to fix that test failure.

File: tests/www/views/test_views_rendered.py [hunk 1]
```diff
@@ -154,13 +154,17 @@ def test_user_defined_filter_and_macros_raise_error(admin_client, create_dag_run
...
         "Dag Serialization is enabled. Hence for the task that have not yet "
         "started running, please use &#39;airflow tasks render&#39; for "
-        "debugging the rendering of template_fields.<br><br>OriginalError: no "
-        "filter named &#39;hello&#39",
-        resp,
-    )
+        "debugging the rendering of template_fields.<br><br>"
+    ) in resp_html
+
+    # MarkupSafe changed the exception detail from 'no filter named' to
```
Comment (potiuk) [inline tests/www/views/test_views_rendered.py:168]: :D

**Review by potiuk — APPROVED** [review]

**Comment by tsingh2k15** [comment]
I am sorry, just looked at the messages, if this change is accepted, I would prefer to completely lift off the upper bound rather than restricting it to `<=2.0`. Is this feasible to lift off the upper bound in separate PR?  Thoughts? Thank you for your support @potiuk @uranusjr

**Comment by potiuk** [comment]
> I am sorry, just looked at the messages, if this change is accepted, I would prefer to completely lift off the upper bound rather than restricting it to `<=2.0`. Is this feasible to lift off the upper bound in separate PR? Thoughts? Thank you for your support @potiuk @uranusjr

Sure - go gor it

**Comment by tsingh2k15** [comment]
> > I am sorry, just looked at the messages, if this change is accepted, I would prefer to completely lift off the upper bound rather than restricting it to `<=2.0`. Is this feasible to lift off the upper bound in separate PR? Thoughts? Thank you for your support @potiuk @uranusjr
> 
> Sure - go gor it

Thanks, I will monitor the test runs for https://github.com/apache/airflow/pull/20113. If there are issues, will close the PR.

**Pull request merged**